### ISSUE TITLE
[BUG] When using Load Balancer IPAM `Pool`s, user is only able to provision 1 LB successfully. both 1st and 2nd LB can't route to the service

### ISSUE THREAD

**Issue by slickwarren** [harvester/harvester#2041]
**Describe the bug**
When using Load Balancer IPAM `Pool`s, user is only able to provision 1 LB successfully. Subsequent LBs using `Pool` will fail
**To Reproduce**
1. deploy harvester and set the `pool CIDR` values so that you can provision LBs using a select range of IPs
2. deploy rancher + connect harvester via `virtualization management`
3. deploy a harvester node driver cluster with cloud provider enabled
4. deploy a workload using a service (load balancer, IPAM = `Pool`)
  * note that the service is working as expected
5. deploy a 2nd workload using a service (load balancer, IPAM = `Pool`)

**Expected behavior**
the 2nd LB will not route to the service correctly, though the service is created and running

**Support bundle**

[supportbundle_ef00de77-99f1-4876-a619-0b061548de01_2022-03-18T17-41-51Z.zip](https://github.com/harvester/harvester/files/8307744/supportbundle_ef00de77-99f1-4876-a619-0b061548de01_2022-03-18T17-41-51Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.1-rc1
 - rancher version v2.6.4-rc9, VM running in managed harvester cluster
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): poweredge r510

related to : https://github.com/rancher/rancher/issues/36956

**Comment by TachunLin** [comment]
@rebeccazzzz, verified **can reproduce** on Rancher `v2.6.4-rc9` with Harvester `v1.0.1-rc`.

### Result
Can provision two Load Balancer with IPAM `Pool` 
  ![image](https://user-images.githubusercontent.com/29251855/160441982-639bf494-f759-4a9f-9d45-d9e9a77e266c.png)


But both Load balancer 1 and 2 are not able to route correctly
* Load Balancer 1
  ![image](https://user-images.githubusercontent.com/29251855/160442748-bc875439-297c-43bd-becf-83943cd4c28a.png)

* Load Balancer 2
  ![image](https://user-images.githubusercontent.com/29251855/160442775-271c1446-ccff-46d1-832b-dcc2b82e8cae.png)

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: v1.0.1-rc1
* Rancher version: Rancher v2.6.4-rc9
* RKE2 version: v1.23.4+rke2r2


### Verify Steps
1. deploy harvester and set the pool CIDR values in settings
  ![image](https://user-images.githubusercontent.com/29251855/160443207-2637d05b-f9e0-4a19-be9e-3b5dc69ddc94.png)
2. deploy rancher + connect harvester via virtualization management
3. deploy a harvester node driver cluster with cloud provider enabled
4. deploy a workload using a service (load balancer, IPAM = Pool)
5. deploy a 2nd workload using a service (load balancer, IPAM = Pool)

**Comment by TachunLin** [comment]
Check the routing connection from harvester node machine

* Unable to connect to route service IP 
```
harvester-one-101rc1:~ # curl http://192.168.122.198/
curl: (7) Failed to connect to 192.168.122.198 port 80: Connection refused
```

* Check the Load balancer service 
  ```
  harvester-one-101rc1:~ # kubectl get svc
  NAME                                            TYPE           CLUSTER-IP      EXTERNAL-IP       PORT(S)          AGE
  kubernetes                                      ClusterIP      10.53.0.1       <none>            443/TCP          8d
  kubernetes-default-lb-test-80ce2223             LoadBalancer   10.53.27.41     192.168.122.75    8080:31442/TCP   7d16h
  rke2-cluster-v1234-default-lb-pool1-26836ee1    LoadBalancer   10.53.218.79    192.168.122.69    8080:30198/TCP   4h1m
  rke2-cluster-v1234-default-lb-test2-bd3f07e9    LoadBalancer   10.53.224.175   192.168.122.90    80:31821/TCP     7d18h
  rke2-cluster2-v1234-default-lb-pool1-5c288b8d   LoadBalancer   10.53.24.219    192.168.122.198   80:30455/TCP     93s
  ```
* Unable to connect to cluster IP 
  ```
  harvester-one-101rc1:~ # curl 10.53.24.219
  curl: (7) Failed to connect to 10.53.24.219 port 80: Connection refused
  ```
  
* The cluster IP is rejected from iptable rule 
  ```
  harvester-one-101rc1:~ # iptables-save | grep 10.53.24.219
  -A KUBE-SERVICES -d 10.53.24.219/32 -p tcp -m comment --comment "default/rke2-cluster2-v1234-default-lb-pool1-5c288b8d has no endpoints" -m tcp --dport 80 -j REJECT --reject-with icmp-port-unreachable
  ```

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: v1.0.1-rc1
* Rancher version: v2.6.4-rc10 
* RKE2 version: v1.23.4+rke2r2
* IP Pool:  192.168.122.0/24

**Comment by yaocw2020** [comment]
Workaround: 
It happens when disabling the health check, so to enable health check as a workaround.

**Comment by TachunLin** [comment]
Verified on Rancher `v2.6.4` with Harvester `v1.0.1-rc2`. The workaround works

### Result 
The workaround to enable health check can make routing correctly for the IPAM Pool Load balancer
![image](https://user-images.githubusercontent.com/29251855/162190420-5f4704be-aed9-4dfb-8161-b0573185d6c7.png)

Enable the health check port
![image](https://user-images.githubusercontent.com/29251855/162186809-e0df4cfa-26d1-4843-ae60-71d98009fbc1.png)

**Comment by yaocw2020** [comment]
Test plan:
Do the steps described in this comment https://github.com/harvester/harvester/issues/2041#issuecomment-1080856111 but replace the cloud provider image as `armarny/harvester-cloud-provider:issue2122` before deploying the load balancer service.

**Comment by TachunLin** [comment]
Verified **fixed** on Harvester v1.0.1 with Rancher v2.6.4. Close this issue.

### Result
On the v1.22.7+rke2r2 cluster in harvester, (local kvm machine)
Not yet require to update the cloud provider image for https://github.com/harvester/harvester/issues/2041#issuecomment-1095091888

1. We can provision the first and second load balancer correctly to the same nginx deployment 
  ![image](https://user-images.githubusercontent.com/29251855/162990914-e01a8fd9-adaf-4887-92d0-e4f009f7ea30.png)


2. Can correctly route to the first and second load balancer
  ![image](https://user-images.githubusercontent.com/29251855/162987173-377629b1-6332-436a-920c-673a792a939e.png)
  ![image](https://user-images.githubusercontent.com/29251855/162986885-fab32ada-8e40-475c-bbc7-a45ef1c5f22e.png)

### Test Information

- Test Environment: 1 node harvester on local kvm machine
- Harvester version: v1.0.1
- Rancher version: v2.6.4
- RKE2 version: v1.22.7+rke2r2
- OS image: ubuntu focal 
- IP Pool: 192.168.122.0/24

### Verify Steps
1. Prepare a one node harvester 
2. Create a ubuntu virtual machine
3. Set the following user data in cloud config
  ```
  password: 123456
  chpasswd: { expire: False }
  ssh_pwauth: True
  ```
4. set the pool CIDR values in settings
  ![image](https://user-images.githubusercontent.com/29251855/162985685-908c08b7-b8e7-451e-ad78-916c07feed51.png)
5. deploy rancher + connect harvester via virtualization management
6. deploy a harvester node driver cluster with cloud provider enabled
7. Deploy a nginx server
8. deploy the 1st workload using a service (load balancer, IPAM = Pool) and apply to the nginx server
  ![image](https://user-images.githubusercontent.com/29251855/162985905-64639731-10de-4a2a-a5d0-113695464a65.png)
  ![image](https://user-images.githubusercontent.com/29251855/162985955-b77239e9-90c5-4381-93bb-4841d1572953.png)
  ![image](https://user-images.githubusercontent.com/29251855/162986040-721d327f-0886-4d5d-a3ac-3b204d2d9fdf.png)

9. deploy a 2nd workload using a service (load balancer, IPAM = Pool) and apply to the nginx server
  ![image](https://user-images.githubusercontent.com/29251855/162986464-d97b1199-d1c0-456f-9242-78b47065c56b.png)
  ![image](https://user-images.githubusercontent.com/29251855/162986551-f1f4856f-2f66-49d4-9886-e18e30829c4d.png)
  
10. Check the route of the 1st load balancer deployment 
11. Check the route of the 2nd load balancer deployment

### Additional Context
** Please be noted **
1. If you encounter the routing works on the 1st load balancer but failed routing at the second
  ![image](https://user-images.githubusercontent.com/29251855/162947022-42764b1c-a52e-4cab-a349-9e70174a06e9.png)
  ![image](https://user-images.githubusercontent.com/29251855/162946932-15a368aa-4e03-43af-b377-fe45ee3c48ba.png)
  
2. This may probably two load balancer get different IP, one of them may have some connectivity restriction
3. We can adjust the IP pool setting in Harvester to make sure assign the same range of IPs to load balancer service 
![image](https://user-images.githubusercontent.com/29251855/162996675-8a0954e9-6233-4d20-a560-53a37779e482.png)

#### Steps to replace your customized image  
1. Edit yaml of the cloud provider deployment in the `Installed Apps`
  ![image](https://user-images.githubusercontent.com/29251855/162997240-bba28af2-e26a-46d8-b8d5-1bd42a34aef0.png)
2. Replace the image value and save 
  ![image](https://user-images.githubusercontent.com/29251855/162997436-f39cc1e3-a970-4516-ae4e-1cd171be89ba.png)

3. Remove the `Running` node, make the `Pending` node work
  ![image](https://user-images.githubusercontent.com/29251855/162998264-28d73cd3-97e3-46ad-9342-d8c75690e78f.png)

4. Replaced load balancer with customized image 
  ![image](https://user-images.githubusercontent.com/29251855/162998908-a87823e1-e031-47d9-bd90-cb52596535f0.png)

**Comment by TachunLin** [comment]
Verified **`fixed`** for KDM release regression test with the following configuration
* Rancher: v2.6.4
* RKE2 version: v1.23.6-rc3+rke2r1
* Cloud provider version: 0.1.1100 

Can create many `Pool` load balancers and route correctly  
 ![image](https://user-images.githubusercontent.com/29251855/164978808-333df3b8-e826-40d3-8c90-d688272fd075.png)
